io/net/irqbalance.py

```diff
@@ -0,0 +1,437 @@
+
+        '''
+        Making CPU offline serially
+        '''
+        for cpus in self.cpu_list[:-1]:
+            self.log.info("Offlining cpu%s", cpus)
+            cpu.offline(cpus)
+            if cpus in cpu.online_list():
+                self.fail(f" The offlined cpu {cpus} is still showing as"
+                          f" online, Please check the logs")
+
+        '''
+        Making CPU online serially
+        '''
+        for cpus in self.cpu_list[:-1]:
+            self.log.info("Onlining cpu%s", cpus)
+            cpu.online(cpus)
+            if cpus not in cpu.online_list():
+                self.fail(f" The onlined cpu {cpus} is still showing as"
+                          f" offline, Please check the logs")
+        dmesg.collect_errors_dmesg(errorlog)
+
+    def test_smt_toggle(self):
+        '''
+        Enables diferrent SMT options to offline multiple cpus
+        1. makes offlines all cpu's
+        2. enable smt value from 1 --> 8
+        3. makes all cpu offline again.
+        4. makes all cpu online.
+
+        '''
+        for i in ['off', *range(1, 9), 'off', 'on']:
+            process.run('ppc64_cpu --smt=%s' % i, shell=True, timeout=15)
+        dmesg.collect_errors_dmesg(errorlog)
+
+    def test_taskset(self):
+        '''
+        Function to run "taskset" command to assign available CPU's to
+        Process ID.
+        changes the CPU number of PID while IO running.
+        Eg:
+           CPU1 ---> CPU2
+           CPU2 ---> CPU3, ----> till last availble CPU number.
+        '''
+        if self.iface:
+            for cpu_number in self.cpu_list:
+                if self.networkinterface.ping_flood(self.interface,
+                                                    self.peer_ip,
+                                                    self.ping_count):
+                    cmd = "taskset -cp %s %s" % (
+                        cpu_number,
+                        self.get_ping_process_pid()
+                    )
+                    output = process.run(cmd, shell=True, ignore_status=True)
+                    if output.exit_status != 0:
+                        self.fail(f'taskset command failed check logs')
+                    if cpu_number != self.taskset_cpu_validation():
+                        self.fail(f"CPU number is mismatching after taskset "
+                                  "operation, Please check logs.")
+                    try:
+                        process.system(
+                            "kill -9 %s" % int(self.get_ping_process_pid()),
+                            ignore_status=True,
+                            sudo=True
+                        )
+                    except CmdError as ex:
+                        self.log.fail(f"Failed to kill the processes, {ex}")
+                else:
+                    self.fail(f"ping flood failed, Please check the logs")
+        if self.disk:
+            for cpu_number in self.cpu_list:
+                if self.dd_run():
```

Comment:
what is the else for this?

what does dd_run() returns here?
I think just running the dd without count will go for more than 10 min
so you can check the rest of the process
please re-check on this

Reply:
@Naresh-ibm 
Added code to exit the script when there is no dd_run happening.
Along with added cpu validation "taskset_cpu_validation()" for block based taskset operation.